view/frontend/templates/js/replacejs.phtml

```diff
@@ -753,85 +755,56 @@ ob_start();
-                        }
-
-                        if (!data.cart.orderToken) {
-                            reject(new Error('The cart is empty.'));
-                            hintBarrier.resolve(hints);
-                            return;
-                        }
+                            cart = {};
+                            hints = {};
+                        } else {
 
                         var prefill = isObject(data.hints.prefill) ? deepMergeObjects(hints.prefill, data.hints.prefill) : hints.prefill;
 
                         hints = deepMergeObjects(hints, data.hints);
                         hints.prefill = prefill;
                         //////////////////////////
-
-                        resolve(data.cart);
-                        hintBarrier.resolve(hints);
+                        }
+                        if (oldBoltCartValue === "") {
+                            // resolve promises for initial .configure() call
+                            cartBarrier.resolve(cart);
+                            hintBarrier.resolve(hints);
+                        } else {
+                            // reconfigure bolt checkout with new values
+                            config = {};
+                            BoltCheckout.configure(cart, hints, callbacks, config);
+                        }
+                        oldBoltCartValue = JSON.stringify(cart);
 
                         // open the checkout if auto-open flag is set
                         // one time only on page load
                         if (initiateCheckout && allowAutoOpen) {
                             BC.open();
                             allowAutoOpen = false;
                         }
+                    });
 
-                        // prefetch Shipping and Tax for multi-step checkout
-                        if (getCheckoutType() === 'checkout') prefetchShipping();
-                    })
-                    .fail(function(jqXHR, textStatus, errorThrown) {
-                        reject(new Error(errorThrown || jqXHR.statusText || 'Network request failed'));
-                    })
-                    .always(function() {
-                        createRequest = false;
-                    })
-            });
-            // Get number of items in cart from counter element
-            var numItems = $(".counter-number").text();
-
-            // Call config with floatingButtonMode if always present checkout enabled and items were or were not just added to cart (dataChanged)
-            var config = boltConfig.always_present_checkout && (numItems && numItems !=="0" || dataChanged)? {floatingButtonMode: dataChanged ? "show_cart" : "show_cart_on_hover" } : {};
-            var BC = BoltCheckout.configure(cart, hintBarrier.promise, callbacks, config);
+            if (isFirstCall) {
+                var BC = BoltCheckout.configure(cartBarrier.promise, hintBarrier.promise, callbacks, boltCheckoutConfig);
```

Comment:
I assume this variable `BC` is not used? if  so let's remove

Reply:
We use it at line 799. Do you think we still need it?
`BC.open();`

Reply:
I don't think above BC.open would work if you do `var BC` here (`var` would define a new variable in this if block scope). maybe what you need is define BC variable at higher level and remove `var` from here

Reply:
Done